### PR TITLE
decouple window_with_time_or_count from observable

### PR DESCRIPTION
**Pull request by gchudnov**
Decoupled `window_with_time_or_count` from observable, fixed a couple of typos.
Please review.

Thank you,
Grigoriy

### CODE REVIEW

File: Rx/v2/src/rxcpp/operators/rx-buffer_time_count.hpp [hunk 1]
```diff
@@ -263,7 +263,7 @@ struct member_overload<buffer_with_time_or_count_tag>
     }
 
     template<class... AN>
-    static operators::detail::buffer_count_invalid_t<AN...> member(AN...) {
+    static operators::detail::buffer_with_time_or_count_tag_invalid_t<AN...> member(AN...) {
         std::terminate();
```
Comment (kirkshoop) [inline Rx/v2/src/rxcpp/operators/rx-buffer_time_count.hpp]: Is _tag_ intended to be in the name?

Comment (gchudnov) [inline Rx/v2/src/rxcpp/operators/rx-buffer_time_count.hpp]: no that's a mistake, should be fixed now...

Comment (kirkshoop) [inline Rx/v2/src/rxcpp/operators/rx-buffer_time_count.hpp]: Ok, thank you!

**Comment by kirkshoop** [comment]
Thank you! There aren't many operators left to decouple! I really appreciate your contributions!

FYI - I have been thinking that some operators like subscribe() and as_dynamic() should always be included even in rx-lite

**Comment by gchudnov** [comment]
got it!
agreed, it rarely makes sense to ask a user to include `subscribe` separately.

**Pull request merged**